src/components/app-wrapper/metadata-helpers/__tests__/visualization.spec.ts

```diff
@@ -0,0 +1,461 @@
+import { describe, it, expect } from 'vitest'
+import {
+    supplementDimensionMetadata,
+    extractTrackedEntityTypeMetadata,
+    extractFixedDimensionsMetadata,
+    extractProgramDimensionsMetadata,
+    extractDimensionMetadata,
+    extractProgramMetadata,
+} from '../visualization'
+import type { SavedVisualization, DimensionRecord } from '@types'
+
+describe('supplementDimensionMetadata', () => {
+    it('should add prefixed dimension metadata for dimensions with matching metadata items', () => {
+        const metadataInput = {
+            dimension1: {
+                uid: 'dimension1',
+                name: 'Dimension One',
+            },
+            dimension2: {
+                uid: 'dimension2',
+                name: 'Dimension Two',
+            },
+        }
+
+        const visualization = {
+            outputType: 'EVENT' as const,
+            columns: [
+                {
+                    dimension: 'dimension1',
+                    dimensionType: 'DATA_ELEMENT',
+                    valueType: 'TEXT',
+                    optionSet: { id: 'optionSet1' },
+                } as DimensionRecord,
+            ],
+            rows: [],
+            filters: [],
+        } as unknown as SavedVisualization
```

Comment:
Probably not that important, but `SavedVisualization` requires `id`.
Can `NewVisualization`/`CurrentVisualization` be used here instead?

Reply:
Since this is a test for metadata extraction from the visualization response, the `SavedVisualization` type is in fact the correct one. I tried if adding an `id` field would enable me to skip the `as unkown` cast, but this object is lacking more fields to be a fully qualified `SavedVisualization`. I also tried `const visualization: Partial<SavedVisualization> = { ... }`, which doesn't cause issues here, but does cause issues when the variable is used as a function parameter of type `SavedVisualization`. Considering that a) we're (re)using the `SavedVisualization` type for the function parameter (and I think this is good practice), and b) we do not need to populate all fieds to successfully test things, the current `as unknown as SavedVisualization` is the most progmatic solution.